[sec] add run configuration for local jetty server

line-o opened the pull request:
The run configuration to start the jetty server within the IDEA IDE that ships with eXist-db exposes the server to the local network (by binding to 0.0.0.0).
A new configuration named "Local Jetty Server" is added which only binds to the host IP `127.0.0.1` aka localhost.
This was made possible by #4196, which adds recognition of the option `jetty.host`.

It is absolutely unnecessary to expose a development instance running inside an IDE to the internet in most cases.

line-o commented:
I do not know if that is an option within the IDEA run configuration. But it is possible to overwrite the default by editing it in the extremely rare case one would need that.

adamretter commented:
Sorry but I am against this. I frequently test across multiple machines when debugging issues.

line-o commented:
While running the db from inside IDEA? @adamretter

adamretter commented:
@line-o Indeed.

line-o commented:
And it is too much to ask to run a proxy for that special use-case?

adamretter commented:
@line-o For me it is the normal use-case not a special use-case.

line-o commented:
I see, how about adding a separate runconf for your use-case?

adamretter commented:
> how about adding a separate runconf for your use-case?

You are assuming this is only my use-case. I suspect others work this way also.

Additionally I don't really understand what the advantage of your PR is over what we currently have?

line-o commented:
The configuration that runs the jetty server only on localhost is now available as a separate option "Local Jetty Server".
I believe this addresses the concerns of you both @ljo and @adamretter

adamretter commented:
I think this is better dealt with my the firewall of your system, in-built on macOS, Linux, and Windows for many years now.

However, as long as this is an additional option and not the default option, then it doesn't effect other developers.
